plugin.py

```diff
@@ -86,10 +86,28 @@ def install_or_update(cls) -> None:
     def markdown_language_id_to_st_syntax_map(cls) -> Optional["MarkdownLangMap"]:
         return {"python": (("python", "py"), ("LSP-pyright/syntaxes/pyright",))}
 
+    def on_server_response_async(self, method: str, response: Response) -> None:
+        if method == "textDocument/hover" and isinstance(response.result, dict):
+            contents = response.result.get("contents")
+            if not isinstance(contents, dict) or not contents.get("kind") == "markdown":
+                return
+            response.result["contents"]["value"] = self.patch_markdown_content(contents["value"])
+
     # -------------- #
     # custom methods #
     # -------------- #
 
+    def patch_markdown_content(self, content: str) -> str:
+        # Add another linebreak before horizontal rule following fenced code block
+        content = re.sub("```\n---", "```\n\n---", content)
+        # Add markup for some common field name conventions in function docstring
+        content = re.sub(
+            r"\n:param[ ]+([\w\\]+):", lambda m: "\n__Param:__ `{}`".format(m.group(1).replace("\\_", "_")), content
```

Comment:
Would `^` work at the beginning instead of `\n`? I believe the latter would not if it it's the first line. (Whether it's realistically possible is another matter).

Reply:
`^` would not work, because the entire Markdown content is a single string.

>  I believe the latter would not if it it's the first line.

I assume it can be ruled out that the field list occurs at the first line of the Markdown content, because Pyright always inserts the function signature in form of a code block on top of the docstring.

Reply:
Right, my whole confusion came from thinking that `\n` is at the start of the match.